### ISSUE TITLE
[EKS] : Apiserver Auditing failed of  request: encoding failed: authentication Error

### ISSUE THREAD

**Issue by oguzhanaygn** [aws/containers-roadmap#1157]
Hi,

I created a AWS EKS 1.18 Cluster and enabled all the control plane logs. When I checked the api server logs on CloudWatch,

**"Auditing failed of  request: encoding failed: authentication.TokenRequest is not suitable for converting to "v1" in scheme "k8s.io/kubernetes/pkg/api/legacyscheme/scheme.go"** logs are coming. But I didn't do any fundamental change on cluster. Can it be a bug?

Thank you.

**Comment by framled** [comment]
I have the same error.

`W1207 14:04:54.006923       1 request.go:157] Auditing failed of  request: encoding failed: authentication.TokenRequest is not suitable for converting to "v1" in scheme "k8s.io/kubernetes/pkg/api/legacyscheme/scheme.go:30"`

I1207 16:01:20.225279       1 log.go:172] http: TLS handshake error from xxx.xxx.xxx.xxx:53052: tls: client offered only unsupported versions: [302 301 300]

**Comment by njuptlzf** [comment]
ref https://github.com/kubernetes/kubernetes/issues/96319

**Comment by njuptlzf** [comment]
ref kubernetes/kubernetes#98612

**Comment by nodesocket** [comment]
👍🏻 same thing. I don't think this is expected.

**Comment by neuralsandwich** [comment]
Same error. Has been in effect for about a week according to my user reports,

**Comment by S4NT14G0** [comment]
I'm seeing this error as well on EKS K8s 1.21 and AMI 1.21.5-20220112.

I have another eks cluster running on K8s 1.21 and AMI 1.21.5-20211206 that works fine

**Comment by niemyjski** [comment]
Same thing

**Comment by nodesocket** [comment]
> I'm seeing this error as well on EKS K8s 1.21 and AMI 1.21.5-20220112.
> 
> I have another eks cluster running on K8s 1.21 and AMI 1.21.5-20211206 that works fine

So, then this is regression? The AMI 20220112 is having this issue but the older AMI version 20211206 is fine?

**Comment by S4NT14G0** [comment]
@nodesocket correct.  To reproduce create a new eks cluster and install ingress-nginx, cert-manager, or external-dns and all three of the pods will fail to start on the newer ami.  The old amo they all work fine though I have seen an skbn cron job fail on the older ami but I'm sure I can work around that.   When I get back to my computer at work I will post some more information to be more thorough.  I'm currently going to spin up a new cluster on 1.20 or 1.19 and install my services on there as a workaround.

**Comment by deepshock-211120** [comment]
Same thing observed on 1.21 with ami 1.21.5-20220112 after enable all types of control plane logging

W0126 09:13:13.702684      11 request.go:161] Auditing failed of  request: encoding failed: authentication.TokenRequest is not suitable for converting to "v1" in scheme "k8s.io/kubernetes/pkg/api/legacyscheme/scheme.go:30"

W0126 09:20:39.436336      11 watcher.go:220] watch chan error: etcdserver: mvcc: required revision has been compacted

**Comment by S4NT14G0** [comment]
@nodesocket I take my earlier comment back.  After logging in and checking on my 1.21 server using the old AMI 1.21.5-20211206 the problem has began to occur.  All of my system services like ingress-nginx, cert-manager, prometheus, & external-dns are now failing (I made no changes since my comment 5 days ago). They are all complaining about not having permission to get K8s API server token.  It seems like this is a bug affecting 1.21 and not specific AMIs.  For everyone else experiencing this issue I'm going to be bringing up new clusters on 1.20 and will report back if it works.  This could be a suitable work around until eks fixes the issues with 1.21

**Comment by mikestef9** [comment]
We need this [change](https://github.com/kubernetes/kubernetes/pull/102040) merged into EKS to stop these log messages. This is included with the upcoming EKS v1.22 release. For now, you can safely ignore the messages.

**Comment by S4NT14G0** [comment]
@mikestef9 Is this message not an indication of a larger problem?  I'm not sure if others are experiencing the same issue in eks, but it seems like the latest update 1.21 is causing issues with the API token being injected into pods as they start up.  Below I attached some logs from a cert-manager pod.  Ever since upgrading to 1.21 I've been having failures across many services with the same error.  I read the k8s release notes and guides to upgrading from 1.19 -> 1.20 -> 1.21 and there doesn't seem to be any indication of changes to the mechanism to allow pods to access token to communicate w/ the K8s api.  The only indicator I have found in my clusters is the error message @framled posted above.

![image](https://user-images.githubusercontent.com/10766443/151229435-29bcdbf8-6b30-4112-b623-91bec6961325.png)
